x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/JobManager.java

```diff
@@ -451,7 +456,7 @@ private void postJobUpdate(UpdateJobAction.Request request, Job updatedJob, Acti
                 ));
             }
         } else {
-            logger.debug("[{}] No process update required for job update: {}", jobUpdate::getJobId, jobUpdate::toString);
+            logger.debug("[{}] No process update required for job update: {}", jobUpdate.getJobId(), jobUpdate.toString());
```

Comment:
Using a supplier means that `jobUpdate.toString()` won't be evaluated unless debug level logging is enabled. `jobUpdate.getJobId()` is trivial but  `jobUpdate.toString()` is not.  What's the reasoning behind this change?

Reply:
You are right. I changed it back.

Reply:
Hm. I think I changed the logger to use elastic's one and now I have the following error:
> error: method debug in interface Logger cannot be applied to given types;
            logger.debug("[{}] No process update required for job update: {}", jobUpdate::getJobId, jobUpdate::toString);
            
But I'll rewrite to prevent premature execution.